### ISSUE TITLE
prefer_single_line_comments does not work for xml

### ISSUE THREAD

**Issue by tkkcc** [b3nj5m1n/kommentary#52]
I set 
```lua
require('kommentary.config').configure_language("default", {
    prefer_single_line_comments = true,
})
```
for
```xml
<string>DejaVu Sans</string>
<string>DejaVu Sans Mono</string>
```
expect(nerdcommenter)
```xml
<!--<string>DejaVu Sans</string>-->
<!--<string>DejaVu Sans Mono</string>-->
```
but get
```xml
<!-- <string>DejaVu Sans</string>
<string>DejaVu Sans Mono</string> -->
```

**Comment by yhu266** [comment]
Do you have `JoosepAlviste/nvim-ts-context-commentstring` installed? The following lines in that plugin's instruction will mess up with `prefer_single_line_comments`
```lua
single_line_comment_string = 'auto',
multi_line_comment_strings = 'auto',
```

**Comment by tkkcc** [comment]
I comment `JoosepAlviste/nvim-ts-context-commentstring` and `suy/vim-context-commentstring`, then do `PlugClean` and `PlugUpdate`. The problem is still. Can you get the expected result? @yhu266

**Comment by yhu266** [comment]
Yes I do get expected results. I have the following configurations (in Lua)

```lua
require('kommentary.config').configure_language('default', {
  prefer_single_line_comments = true,
  use_consistent_indentation = true,
  ignore_whitespace = true,
  -- JoosepAlviste/nvim-ts-context-commentstring
  hook_function = function()
    require('ts_context_commentstring.internal').update_commentstring()
  end,
})
```

**Comment by tkkcc** [comment]
Thanks @yhu266 . I tried but problem remains. Because xml's commentstring `<!--%s-->` does not end with `%s`, the following function believes `xml` is a language that doesn't support multi-line comments. The ending `return` gives `{false,{"<!--","-->"}}`.

https://github.com/b3nj5m1n/kommentary/blob/0f8e910156b8ea92690ec3c85808cdba04007ed0/lua/kommentary/config.lua#L269-L283

This will turn `mode` into `force_multi`, and ignore user options.

https://github.com/b3nj5m1n/kommentary/blob/0f8e910156b8ea92690ec3c85808cdba04007ed0/lua/kommentary/config.lua#L377-L384

It is `safe` but not `perfect`. Also, current `single_line_comment_string` does not support surrounding comment.

**Comment by Iron-E** [comment]
The same thing is happening for me on Markdown and HTML filetypes.

> Do you have JoosepAlviste/nvim-ts-context-commentstring installed? The following lines in that plugin's instruction will mess up with prefer_single_line_comments

For context, I don't have this installed.
